src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesManagedTransform.Browser.cs

```diff
@@ -0,0 +1,1031 @@
+            if (disposing)
+            {
+                // We need to always zeroize the following fields because they contain sensitive data
+                if (_IV != null)
+                {
+                    Array.Clear(_IV);
+                    _IV = null!;
+                }
+                if (_lastBlockBuffer != null)
+                {
+                    Array.Clear(_lastBlockBuffer);
+                    _lastBlockBuffer = null!;
+                }
+                if (_encryptKeyExpansion != null)
+                {
+                    Array.Clear(_encryptKeyExpansion);
+                    _encryptKeyExpansion = null!;
+                }
+                if (_decryptKeyExpansion != null)
+                {
+                    Array.Clear(_decryptKeyExpansion);
+                    _decryptKeyExpansion = null!;
+                }
+            }
+
+            base.Dispose(disposing);
+        }
+
+        public override int Transform(ReadOnlySpan<byte> input, Span<byte> output)
+        {
+            Debug.Assert(input.Length % BlockSizeBytes == 0);
+            Debug.Assert(output.Length >= input.Length);
+
+            // the below algorithm doesn't allow overlap, so rent a buffer to transform into
+            if (input.Overlaps(output, out int offset) && offset != 0)
+            {
+                byte[] rented = CryptoPool.Rent(input.Length);
+                int bytesWritten = 0;
+
+                try
+                {
+                    bytesWritten = _encrypting ?
+                        EncryptData(input, rented) :
+                        DecryptData(input, rented);
+                    rented.AsSpan(0, bytesWritten).CopyTo(output);
+                    return bytesWritten;
+                }
+                finally
+                {
+                    CryptoPool.Return(rented, clearSize: bytesWritten);
+                }
+            }
+            else
+            {
+                // with no overlap, we can just write directly to the output
+                return _encrypting ?
+                    EncryptData(input, output) :
+                    DecryptData(input, output);
+            }
+        }
+
+        public override int TransformFinal(ReadOnlySpan<byte> input, Span<byte> output)
+        {
+            int bytesWritten = Transform(input, output);
+            Reset();
+            return bytesWritten;
+        }
+
+        //
+        // resets the state of the transform
+        //
+
+        void ILiteSymmetricCipher.Reset(ReadOnlySpan<byte> iv) => throw new NotImplementedException(); // never invoked
+
+        private void Reset()
+        {
+            _IV.AsSpan().CopyTo(_lastBlockBuffer);
+        }
+
+        //
+        // Encrypts the inputBuffer into the outputBuffer using the AES encryption routine.
+        // This method writes the encrypted data into the output buffer.
+        //
+        private int EncryptData(ReadOnlySpan<byte> inputBuffer, Span<byte> outputBuffer)
```

Comment:
Toss-up: Should we call them `source` and `destination` to match span style, or `inputBuffer` and `outputBuffer` to show their NetFX heritage?

Reply:
ICryptoTransform calls them `inputBuffer` and `outputBuffer`.

BasicSymmetricCipher calls them `input` and `output`.

I guess I can drop the `Buffer` part and just go with what the base class calls them.